server/src/child.rs

```diff
@@ -15,20 +15,23 @@ pub struct Thread {
     pub thread: JoinHandle<()>,
     pub tx: SyncSender<ServerMessage>,
     pub rx: Receiver<DebugMessage>,
-
+    
     pub execution: Option<i32>,
     pub id: i32,
 }
 
+
+
 /// messages the debug event loop sends to the server
 pub enum DebugMessage {
     Attached,
     Functions(Vec<Function>),
+    Execution(Exec),
     Function(Function),
     Breakpoints(Vec<usize>),
     Breakpoint,
     BreakpointRemoved,
-    Executing,
+    Executing(Exec),
     Trace(DebugTrace),
     Error(io::Error),
 }
```

Comment:
Preferably the debug thread should not need to worry about the execution info endpoints, so `DebugMessage::Execution` should go away and `DebugMessage::Executing` should stay parameterless.